File: MemExamp.cpp
#include <iostream>

int sumNumbers(int* arr, int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum += arr[i];
    }
    return sum;
}

int main() {
    int* num = new int[5]{10,20,30,40,50}; // Heap allocation
    std::cout << "Sum: " << sumNumbers(num, 5) << std::endl;
    
    delete[] num; // Manual deallocation (avoids memory leak)
    return 0;
}

File: memExamp.rs
fn main() {
    let numbers = vec![1, 2, 3, 4, 5]; // Heap-allocated vector (Rust automatically frees it)
    println!("Numbers: {:?}", numbers);

    let sum = sum_numbers(&numbers); // Borrowing (no ownership transfer)
    println!("Sum: {}", sum);
} 

fn sum_numbers(arr: &Vec<i32>) -> i32 {
    arr.iter().sum()
} // arr is borrowed, not moved, so it is not freed here
